datasets/cn/peoples_congress_wikipedia/crawler.py

```diff
@@ -119,6 +161,60 @@ def parse_table(
         yield row
 
 
+def parse_table_with_rowspan(
+    context: Context,
+    table: HtmlElement,
+) -> Generator[Dict[str, HtmlElement], None, None]:
+    headers = None
+    rowspan_cells = {}  # Dictionary to keep track of cells with rowspan
+
+    for row_index, row in enumerate(table.findall(".//tr")):
+        if headers is None:
+            headers = []
+            for el in row.findall("./th"):
+                # Workaround because lxml-stubs doesn't yet support HtmlElement
+                eltree = el  # If necessary, you can use cast(HtmlElement, el)
+                label = strip_note(eltree.text_content())
+                headers.append(context.lookup_value("headers", label))
+            continue
+
+        cells = row.findall("./td")
+        row_data = {}
+
+        cell_index = 0  # To track which header corresponds to which cell
+        for idx, cell in enumerate(cells):
+            while cell_index in rowspan_cells and rowspan_cells[cell_index][1] > 0:
```

Comment:
I'm not sure I understand this - why two loops? Can you explain how it works in words, perhaps with excel style cell references for the example?

    for each cell in the row
        for each what?

Reply:
Perhaps the general mechanism could be documented in the function-level docstring?